sigar_aix.go

```diff
@@ -0,0 +1,538 @@
+			fs.SysTypeName = "nfs"
+		case C.MNT_JFS:
+			fs.SysTypeName = "jfs"
+		case C.MNT_CDROM:
+			fs.SysTypeName = "cdrom"
+		case C.MNT_PROCFS:
+			fs.SysTypeName = "proc"
+		case C.MNT_SFS:
+			fs.SysTypeName = "sfs"
+		case C.MNT_CACHEFS:
+			fs.SysTypeName = "cachefs"
+		case C.MNT_NFS3:
+			fs.SysTypeName = "nfs3"
+		case C.MNT_AUTOFS:
+			fs.SysTypeName = "autofs"
+		case C.MNT_POOLFS:
+			fs.SysTypeName = "poolfs"
+		case C.MNT_UDF:
+			fs.SysTypeName = "udfs"
+		case C.MNT_NFS4:
+			fs.SysTypeName = "nfs4"
+		case C.MNT_CIFS:
+			fs.SysTypeName = "cifs"
+		case C.MNT_PMEMFS:
+			fs.SysTypeName = "pmemfs"
+		case C.MNT_AHAFS:
+			fs.SysTypeName = "ahafs"
+		case C.MNT_STNFS:
+			fs.SysTypeName = "stnfs"
+		default:
+			if ent.vmt_flags&C.MNT_REMOTE != 0 {
+				fs.SysTypeName = "network"
+			} else {
+				fs.SysTypeName = "none"
+			}
+		}
+
+		fs.DirName = convertBytesToString(vmt2data(buf, ent, C.VMT_STUB, entOff))
+		fs.Options = convertBytesToString(vmt2data(buf, ent, C.VMT_ARGS, entOff))
+		devname := convertBytesToString(vmt2data(buf, ent, C.VMT_OBJECT, entOff))
+		if ent.vmt_flags&C.MNT_REMOTE != 0 {
+			hostname := convertBytesToString(vmt2data(buf, ent, C.VMT_OBJECT, entOff))
+			fs.DevName = hostname + ":" + devname
+		} else {
+			fs.DevName = devname
+		}
+
+		entOff += int(ent.vmt_length)
+	}
+
+	self.List = fslist
+
+	return nil
+}
+
+func (self *LoadAverage) Get() error {
+	cpudata := C.perfstat_cpu_total_t{}
+
+	if _, err := C.perfstat_cpu_total(nil, &cpudata, C.sizeof_perfstat_cpu_total_t, 1); err != nil {
+		return fmt.Errorf("perfstat_cpu_total: %s", err)
+	}
+
+	// from libperfstat.h:
+	// "To calculate the load average, divide the numbers by (1<<SBITS).
+	//  SBITS is defined in <sys/proc.h>."
+	fixedToFloat64 := func(x uint64) float64 {
+		return float64(x) / (1 << C.SBITS)
+	}
+	self.One = fixedToFloat64(uint64(cpudata.loadavg[0]))
+	self.Five = fixedToFloat64(uint64(cpudata.loadavg[1]))
+	self.Fifteen = fixedToFloat64(uint64(cpudata.loadavg[2]))
+
+	return nil
+
+}
+
+func (self *Uptime) Get() error {
+	uptime := time.Now().Sub(time.Unix(int64(system.btime), 0))
+	self.Length = uptime.Seconds()
+	return nil
+}
+
+func (self *Mem) Get() error {
+	meminfo := C.perfstat_memory_total_t{}
+	_, err := C.perfstat_memory_total(nil, &meminfo, C.sizeof_perfstat_memory_total_t, 1)
+	if err != nil {
+		return fmt.Errorf("perfstat_memory_total: %s", err)
+	}
+
+	self.Total = uint64(meminfo.real_total) * system.pagesize
+	self.Free = uint64(meminfo.real_free) * system.pagesize
+
+	kern := uint64(meminfo.numperm) * system.pagesize // number of pages in file cache
+
+	self.Used = self.Total - self.Free
+	self.ActualFree = self.Free + kern
+	self.ActualUsed = self.Used - kern
+
+	return nil
+}
+
+func (self *Swap) Get() error {
+	ps := C.perfstat_pagingspace_t{}
+	id := C.perfstat_id_t{}
+
+	id.name[0] = 0
+
+	for {
+		if _, err := C.perfstat_pagingspace(&id, &ps, C.sizeof_perfstat_pagingspace_t, 1); err != nil {
+			return nil
```

Comment:
Why are you ignoring this error?

Reply:
Good question ! That was a mistake I think.